client/src/modules/cohort/model/actions.js

```diff
@@ -410,16 +410,26 @@ export const leaveCohort = (cohortId, userId, userName) => dispatch =>
       });
     });
 
-export const lockCohortHeader = (cohortId, { nameLock, descriptionLock }) =>
+export const lockCohortHeader = (
+  cohortId,
+  userId,
+  { nameLock, descriptionLock }
+) =>
   socket.emit(CohortHeaderStatusTypes.LOCK, {
     cohortId,
     descriptionLock,
-    nameLock
+    nameLock,
+    userId
   });
 
-export const unlockCohortHeader = (cohortId, { nameLock, descriptionLock }) =>
+export const unlockCohortHeader = (
+  cohortId,
+  userId,
+  { nameLock, descriptionLock }
+) =>
   socket.emit(CohortHeaderStatusTypes.UNLOCK, {
```

Comment:
Consider using a single locking mechanism that locks particular `resourceId` and is handled in a single place. Then, this can be used to handle name and description locks of multiple things like cohorts and list in a coherent way.

Reply:
Saved to a separated task in Jira for coming sprints